clients/drcachesim/scheduler/scheduler_impl.cpp

```diff
@@ -735,7 +740,7 @@ scheduler_impl_tmpl_t<RecordType, ReaderType>::init(
                 input_info_t &input = inputs_.back();
                 input.index = index;
                 input.workload = workload_idx;
-                workload2inputs[workload_idx].push_back(index);
+                workloads_.back().inputs.push_back(index);
```

Comment:
Maybe add to workloads_ later when we actually have the input list, so that the inputs can be set in workload_info_t at construction time and we can mark that vector const?

Reply:
OK, created a temporary local vector and made the workload field const and used std::move to try to avoid the copying costs.